src/index/LocatedTriples.cpp

```diff
@@ -0,0 +1,243 @@
+// Copyright 2023 - 2024, University of Freiburg
+// Chair of Algorithms and Data Structures
+// Authors:
+//    2023 Hannah Bast <bast@cs.uni-freiburg.de>
+//    2024 Julian Mundhahs <mundhahj@tf.uni-freiburg.de>
+
+#include "index/LocatedTriples.h"
+
+#include <algorithm>
+
+#include "absl/strings/str_join.h"
+#include "index/CompressedRelation.h"
+#include "index/IndexMetaData.h"
+#include "index/Permutation.h"
+
+// ____________________________________________________________________________
+std::vector<LocatedTriple> LocatedTriple::locateTriplesInPermutation(
+    const std::vector<IdTriple<0>>& triples,
+    const std::vector<CompressedBlockMetadata>& blockMetadata,
+    const std::array<size_t, 3>& keyOrder, bool shouldExist,
+    ad_utility::SharedCancellationHandle cancellationHandle) {
+  vector<LocatedTriple> out;
+  out.reserve(triples.size());
+  size_t currentBlockIndex;
+  for (auto triple : triples) {
+    triple = triple.permute(keyOrder);
+    currentBlockIndex =
+        std::ranges::lower_bound(
+            blockMetadata, triple.toPermutedTriple(),
+            [&](const CompressedBlockMetadata::PermutedTriple& block,
+                const CompressedBlockMetadata::PermutedTriple& triple) {
+              return block < triple;
+            },
```

Comment:
```suggestion
            std:less<>{},
```

And comment again the rule which this enforces.